lms/static/js/student_account/register.js

```diff
@@ -0,0 +1,111 @@
+                var iframe_head = $(this).contents().find('head');
+                // overwrite styles in child page
+                // hide top navigation and footer
+                var style = $('<style>', {'type': 'text/css'});
+                style.html(`
+                    .nav-skip, header#global-navigation, .wrapper-footer {
+                        display:none;
+                    }
+                    .container.about {
+                        padding: 0;
+                        min-width: auto;
+                    }
+                `);
+                iframe_head.append(style);
+                // set the iframe's height to fill the available space
+                $(this).css({
+                    height: $(content_selector).height()
+                });
+            }
+        });
+    };
+    var showModal = function(modal_selector) {
+        $(modal_selector).show();
+        // prevent scrolling of the background
+        $('body').addClass('open-modal');
+    };
+    var hideModal = function(modal_selector) {
+        $(modal_selector).hide();
+        $('body').removeClass('open-modal');
+        $(modal_selector).find('iframe').remove();
+    };
+    $(document).ready(function() {
+        var tos_link_selector = '.checkbox-honor_code .supplemental-link a',
+            close_button_class = 'modal-close-button',
+            close_button_selector = '.' + close_button_class,
+            content_class = 'modal-content',
+            content_selector = '.' + content_class,
+            modal_class = 'modal',
+            modal_selector = '.' + modal_class;
+        $('body').on('click', tos_link_selector, function(event) {
+            var link = event.target;
+            event.preventDefault();
+            // ignore disabled TOS
+            if (link.href.endsWith('#')) {
+                return;
+            }
+            if ($(modal_selector).length < 1) {
+                // add the modal if it doesn't exist yet
+                $('body').append(buildModal(link, modal_class, content_class, close_button_class));
+            }
+            // add a new iframe to the content area
+            $(content_selector).append(buildIframe(link, content_selector));
+            showModal(modal_selector);
+            $(close_button_selector).focus();
+        });
+        $('body').on('click', close_button_selector, function(event) {
+            hideModal(modal_selector);
+        });
+        // hide the modal when clicking outside its content
+        $('body').on('click', modal_selector, function(event) {
+            if ($(event.target).hasClass(modal_class)) {
+                hideModal(modal_selector);
+            }
+        });
+        $(document).keydown(function (event) {
+            if ((event.keyCode === 27) && ($(modal_selector).is(':visible'))) {
+                // hide the modal if ESC is pressed
+                hideModal(modal_selector);
+                event.preventDefault();
```

Comment:
@replaceafill Any specific reason for calling `preventDefault` last (instead of calling it before any other logic)?